awserr.go

```diff
@@ -33,5 +33,12 @@ func IsAWSErrExtended(err error, code string, message string, origErrMessage str
 	if !IsAWSErr(err, code, message) {
 		return false
 	}
-	return strings.Contains(err.(awserr.Error).OrigErr().Error(), origErrMessage)
+
+	// Ensure OrigErr() is non-nil, to prevent panics
+	if origErr := err.(awserr.Error).OrigErr(); origErr != nil {
+		return strings.Contains(origErr.Error(), origErrMessage)
+	}
+
+	// Allow missing OrigErr() with missing origErrMessage
+	return origErrMessage == ""
```

Comment:
The conditional flow here seems odd to me, but maybe it's idiomatic for Go. If origErrMessage is missing, couldn't you exit there, then do the Contains check?

Reply:
I believe so. I'll submit a followup commit to flip this around.